oct-memcached-preprovision.go

```diff
@@ -1,162 +1,157 @@
+		cachenodetype = os.Getenv("LARGE_INSTANCE_TYPE")
+		numcachenodes = int64(1)
+	}
 
-    svc := elasticache.New(session.New(&aws.Config{
-        Region:      aws.String("us-west-2"),
-    } ))
-
-
-
-    params := &elasticache.CreateCacheClusterInput{
-	CacheClusterId:          aws.String(name),
-	AutoMinorVersionUpgrade: aws.Bool(true),
-	CacheNodeType:           aws.String(cachenodetype),
-	CacheParameterGroupName: aws.String(cacheparametergroupname),
-	CacheSubnetGroupName: aws.String("memcached-subnet-group"),
-	Engine:               aws.String("Memcached"),
-	NumCacheNodes:        aws.Int64(numcachenodes),
-	Port:                 aws.Int64(6379),
-	SecurityGroupIds: []*string{
-		aws.String(os.Getenv("ELASTICACHE_SECURITY_GROUP")),
-	},
-	Tags: []*elasticache.Tag{
-		{
-			Key:   aws.String("Name"),
-			Value: aws.String(name),
+	svc := elasticache.New(session.New(&aws.Config{
+		Region: aws.String(os.Getenv("REGION")),
+	}))
+
+	params := &elasticache.CreateCacheClusterInput{
+		CacheClusterId:          aws.String(name),
+		AutoMinorVersionUpgrade: aws.Bool(true),
+		CacheNodeType:           aws.String(cachenodetype),
+		CacheParameterGroupName: aws.String(cacheparametergroupname),
+		CacheSubnetGroupName:    aws.String("memcached-subnet-group"),
+		Engine:                  aws.String("Memcached"),
+		NumCacheNodes:           aws.Int64(numcachenodes),
+		Port:                    aws.Int64(6379),
+		SecurityGroupIds: []*string{
+			aws.String(os.Getenv("ELASTICACHE_SECURITY_GROUP")),
+		},
+		Tags: []*elasticache.Tag{
+			{
+				Key:   aws.String("Name"),
+				Value: aws.String(name),
+			},
+			{
+				Key:   aws.String("billingcode"),
+				Value: aws.String(billingcode),
+			},
 		},
-                {
-                        Key:   aws.String("billingcode"),
-                        Value: aws.String(billingcode),
-                },
-	},
-      }
-    resp, err := svc.CreateCacheCluster(params)
-
-    if err != nil {
-	fmt.Println(err.Error())
-	return err.Error()
-    }
-
-    fmt.Println(resp)
-    return name
+	}
+	resp, err := svc.CreateCacheCluster(params)
+
+	if err != nil {
+		fmt.Println(err.Error())
+		return err.Error()
+	}
+
+	fmt.Println(resp)
+	return name
 }
 
-func insertnew(name string, plan string, claimed string){
-     uri:=os.Getenv("BROKER_DB")
-     db, err := sql.Open("postgres", uri)
-        if err != nil {
-                  fmt.Println(err)
-                  os.Exit(2)
-        }
-     var newname string
-     err = db.QueryRow("INSERT INTO provision(name,plan,claimed) VALUES($1,$2,$3) returning name;", name, plan, claimed).Scan(&newname)
-
-        if err != nil {
-                  fmt.Println(err)
-                  os.Exit(2)
-        }
-     fmt.Println(newname)
-     err =db.Close()
+func insertnew(name string, plan string, claimed string) {
+	uri := os.Getenv("BROKER_DB")
+	db, err := sql.Open("postgres", uri)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(2)
+	}
+	var newname string
```

Comment:
The sql connection will leak if the connection isn't closed when this function is returned, we close it on line 96 but its possible during an error for it to exit out on line 93, can you remove line 96 and above here add a `defer db.Close()`, that will ensure the sql connection closes regardless of what path the code takes below it.